flaml/automl.py

```diff
@@ -482,15 +485,16 @@ def _validate_data(self, X_train_all, y_train_all, dataframe, label,
     def _prepare_data(self,
                       eval_method,
                       split_ratio,
-                      n_splits):
+                      n_splits,
+                      period=None):
         X_val, y_val = self._state.X_val, self._state.y_val
         if issparse(X_val):
             X_val = X_val.tocsr()
         X_train_all, y_train_all = \
             self._X_train_all, self._y_train_all
         if issparse(X_train_all):
             X_train_all = X_train_all.tocsr()
-        if self._state.task != 'regression' and self._state.fit_kwargs.get(
+        if self._state.task == 'classification' and self._state.fit_kwargs.get(

```

Comment:
`self._state.task` can be 'binary:logistic` or `multi:softmax`. See https://github.com/microsoft/FLAML/blob/10082b9262a862ef51cf09117bc5d13e43d1ca0d/flaml/automl.py#L1107